## Getting started
### Clone the repo to your working directory

```
git clone https://github.com/tanyaphung/vcfhelper.git
```

### Version and dependencies
* vcfhelper was developed and tested using Python version 3.7

### About
* vcfhelper is a collection of python scripts to help with processing vcf files. vcfhelper contains scripts to:
    1. Extract annotations from the vcf file for each variant
    2. Tabulate the number of sites in the vcf 

In [6]:
## Change to the relevant directory
import os
os.getcwd()
os.chdir('c:\\Users\\tuyen\\Documents\\github_repo\\vcfhelper')

## Usage
### Extract annotations from the vcf file for each variant
* For this, call the script `extract_stats_from_vcf.py`

In [7]:
%run extract_stats_from_vcf.py -h

usage: extract_stats_from_vcf.py [-h] --vcf VCF --outfile OUTFILE S [S ...]

This script extracts statistics from vcf file.

positional arguments:
  S                  Statistics from the vcf file to be extracted.

optional arguments:
  -h, --help         show this help message and exit
  --vcf VCF          REQUIRED. Path to the vcf file.
  --outfile OUTFILE  REQUIRED. Path to the output file.


* Example: Let's say that I want to extract out the depth information (DP) and fisher score (FS). This is how I would run the script:

In [7]:
%run extract_stats_from_vcf.py DP FS --vcf test_files/chrX.vcf --outfile test_files/chrX_DP_FS.txt

In [8]:
%%bash 
head test_files/chrX_DP_FS.txt

DP	FS
349	0.000
395	11.633
646	0.000
567	0.000
600	2.808
600	0.000
609	2.310
640	2.347
646	2.555


### Calculate the number of sites/variants in a vcf
* For this, call the script `calc_num_site_in_vcf.py`

In [9]:
%run calc_num_site_in_vcf.py -h

usage: calc_num_site_in_vcf.py [-h] --vcf VCF

Calculate the number of sites in a vcf file

optional arguments:
  -h, --help  show this help message and exit
  --vcf VCF   REQUIRED. Input the path to the vcf file.


* Example: Let's say that I want to calculate how many sites there are in the vcf file chrX.vcf in the test_files directory. This is how I run the script:

In [10]:
%run calc_num_site_in_vcf.py --vcf test_files/chrX.vcf

471980


* This indicates that there are 471,980 sites in the vcf file. 